apps/generator-canonical-ds/src/component/templates/Component.tsx.ejs

```diff
@@ -0,0 +1,32 @@
+/* <%= generatorPackageName %> <%= generatorPackageVersion %> */
+import React from 'react'
+import type { <%= componentName %>Props } from './types.js'
+
+<% if (includeStyles) { %>
+import './<%= componentName %>.css'
+<% } %>
+  
+/**
+ * description of the <%= componentName %> component
+ * @returns {React.ReactElement} - Rendered <%= componentName %>
+ */
+export default function <%= componentName %>({
+  children,
+  className,
+  ...props
```

Comment:
Yes, especially TypeScript doesn't easily allow spreading.

As far as I understand, in setup like currently (I had the same problem with initial implementation of the Button), TS compiler would not allow any additional props that are not listed in `[Component]Props` type.

So for spread to be allowed, the type needs to allow for additional props. This is a bit clunky in TS, as usually it requires a union with a specific type of a specific HTML element, so that TS will know what kind of props are allowed.

Example implementation for Button component using `<button>` props:
https://github.com/canonical/ds25/blob/e4163296ea952215712709887c30d8987444d10e/packages/ds-react-core/src/ui/Button/Button.tsx#L29-L30

Reply:
`id` and `style` have been added by default.

The other props seem a bit component-specific, I don't think they should be included by default.